packages/native-stack/src/views/NativeStackView.native.tsx

```diff
@@ -236,6 +237,13 @@ const SceneView = ({
   const [customHeaderHeight, setCustomHeaderHeight] =
     React.useState(defaultHeaderHeight);
 
+  const cachedAnimatedHeaderHeight = React.useRef(customHeaderHeight);
+  const animatedHeaderHeight = React.useRef(
+    new Animated.Value(customHeaderHeight, {
+      useNativeDriver: true,
```

Comment:
...and you don't need that in a value

Reply:
I didn't know that 😄
Resolved in https://github.com/react-navigation/react-navigation/pull/11663/commits/f7947728c3bab59c1d88585d3c819d85bf540f44.